pkg/fleet/fleet.go

```diff
@@ -0,0 +1,41 @@
+package fleet
+
+import (
+	"errors"
+	"fmt"
+
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// GetClusterHost returns the API server host for the specified client configuration.
+func GetClusterHost(clientCfg clientcmd.ClientConfig) (string, error) {
+	icc, err := rest.InClusterConfig()
+	if err == nil {
+		return icc.Host, nil
+	}
+
+	fail := func(err error) (string, error) {
```

Comment:
Probably better to wrap these errors `fmt.Errorf("...: %w")` rather than using `%v` here ?

https://go.dev/blog/go1.13-errors#wrapping-errors-with-w

Reply:
Much better, thanks for bringing this up. Fixed.